truffle/src/main/java/org/jruby/truffle/language/constants/LookupConstantWithLexicalScopeNode.java

```diff
@@ -49,6 +50,9 @@ protected RubyConstant lookupConstant(VirtualFrame frame,
         if (!isVisible) {
             throw new RaiseException(coreExceptions().nameErrorPrivateConstant(getModule(), name, this));
         }
+        if(constant != null && constant.isDeprecated()){
+            warnDeprecatedConstant(name);
+        }
```

Comment:
Should anything be done here performance wise? A binary profile? Or something else?